drivers/syscon/CMakeLists.txt

```diff
@@ -1,4 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_library()
-zephyr_library_sources_ifdef(CONFIG_SYSCON          syscon.c)
+if(DEFINED CONFIG_SYSCON_GENERIC)
```

Comment:
I believe this is no longer needed, as we already have this:
https://github.com/zephyrproject-rtos/zephyr/blob/ef0d955758e549f39d97b07089dedc4bdd05bc0e/drivers/CMakeLists.txt#L61

and you are now defaulting the `CONFIG_SYSCON` based on devicetree.

It is ok for a user enabling `CONFIG_SYSCON` but not adding any drivers to the library to see the warning, as it actually means there is a mismatch in the config settings.
```suggestion
```

Reply:
Done